diff-test/core/src/sanity.ts

```diff
@@ -0,0 +1,51 @@
+import _ from 'underscore';
+import { P, C, TRUSTING_SECONDS } from './constants.js';
+
+/**
+ * Checks that the model does not overapproximate the system
+ * too much, as this would make it less useful for generating tests.
+ *
+ * For example, we should not ever
+ *    - allow an empty validator set
+ *    - cause a tendermint light client to expire
+ */
+class Sanity {
+  // the timestamp contained in the latest trusted header
+  tLastTrustedHeader = _.object([
+    [P, 0],
+    [C, 0],
+  ]) as { provider: number; consumer: number };
```

Comment:
Why not just write `{P: 0, C: 0}`?

Reply:
This would create an object mapping the characters 'P' to 0 and 'C' to 0 but I want the words 'provider', 'consumer'.